src/api/helpers.jl

```diff
@@ -174,6 +174,22 @@ end
     end
 end
 
+@static if v"1.12.3" ≤ _libhdf5_build_ver
+    function h5d_chunk_iter(dset_id, dxpl_id, cb::Function, op_data::T=nothing) where T
```

Comment:
Probably better to define a "helper" cfunction and pass the function itself through the `op_data` arg (similar to how the other iterator functions work (e.g. `h5a_iterate`), as it can be used on platforms (e.g. ARM) where cfunction closures don't work

Reply:
Ah, I see.

Reply:
Resolved in [3a69b03](https://github.com/JuliaIO/HDF5.jl/pull/1031/commits/3a69b03389b2762f0f85bab80fae98c4ee83396f)